(docs): update envoy plugin docs to match new configurable grpc msg sizes

emaincourt opened the pull request:
Signed-off-by: Elliot Maincourt <e.maincourt@gmail.com>

<!--

Thanks for submitting a PR to OPA!

Before pressing 'Create pull request' please read the checklist below.

* All code changes should be accompanied with tests. If you are not
modifying any tests, just provide a short explanation of why updates
to tests are not necessary. In addition to helping catch bugs, tests
are extremely helpful in providing _context_ that explains how your
changes can be used.

* All changes to public APIs **must** be accompanied with
docs. Examples of public APIs include built-in functions,
config fields, and of course, exported Go types/functions/constants/etc.

* Commit messages should explain _why_ you made the changes, not what
you changed. Use active voice. Keep the subject line under 50
characters or so.

* All commits must be signed off by the author. If you are not
familiar with signing off, see our contributor guide below.

For more information on contributing to OPA see:

* [Contributing Guide](https://www.openpolicyagent.org/docs/latest/contributing/)
  for high-level contributing guidelines and development setup.

-->

Hi,

This PR updates documentation for `opa-envoy-plugin` to match this [change](https://github.com/open-policy-agent/opa-envoy-plugin/pull/323) that allows to configure GRPC server's max rcv/send message sizes.

Thanks in advance for reviewing

srenatus commented on docs/content/envoy-introduction.md:
```diff
@@ -49,6 +49,8 @@ The OPA-Envoy plugin supports the following configuration fields:
 | `plugins["envoy_ext_authz_grpc"].enable-reflection` | No       | Enables gRPC server reflection on the Envoy External Authorization gRPC server. Default: `false`.                                                                                                                                                                                                                                            ...
 | `plugins["envoy_ext_authz_grpc"].proto-descriptor`  | No       | Set the path to a pb that enables the capability to decode the raw body to the parsed body. Default: turns this capability off.                                                                                                                                                                                                              ...
+| `plugins["envoy_ext_authz_grpc"].grpc-max-recv-msg-size` | No       | Set the max message size in bytes the GRPC server can receive. Defaults to 4MB.                                                                                                                                                                                                                                                         ...
+| `plugins["envoy_ext_authz_grpc"].grpc-max-send-msg-size` | No       | Set the max message size in bytes the GRPC server can send. Defaults to `math.MaxInt32`.                                                                                                                                                                                                                                                ...
```
[nit] I'm not sure everyone knows roughly what `math.MaxInt32` means in this context. Can we spell it out, or give some other indication, please?

emaincourt replied:
Sure, I've replaced it with a more explicit description.

srenatus reviewed (commented):
thanks for adding this. gave it a quick review, some notes 👇

emaincourt commented on docs/content/envoy-introduction.md:
```diff
@@ -49,6 +49,8 @@ The OPA-Envoy plugin supports the following configuration fields:
 | `plugins["envoy_ext_authz_grpc"].dry-run`           | No       | Configures the Envoy External Authorization gRPC server to unconditionally return an `ext_authz.CheckResponse.Status` of `google_rpc.Status{Code: google_rpc.OK}`. Default: `false`.                                                                                                                                                         ...
 | `plugins["envoy_ext_authz_grpc"].enable-reflection` | No       | Enables gRPC server reflection on the Envoy External Authorization gRPC server. Default: `false`.                                                                                                                                                                                                                                            ...
 | `plugins["envoy_ext_authz_grpc"].proto-descriptor`  | No       | Set the path to a pb that enables the capability to decode the raw body to the parsed body. Default: turns this capability off.                                                                                                                                                                                                              ...
+| `plugins["envoy_ext_authz_grpc"].grpc-max-recv-msg-size` | No       | Set the max message size in bytes the GRPC server can receive. Defaults to 4MB.                                                                                                                                                                                                                                                         ...
```
Sorry, I should have caught this one.

emaincourt commented:
@srenatus Thanks a lot for your review. I should have addressed all the comments. Does it look better to you now ?

srenatus reviewed (dismissed):
Thanks. I take it we'll wait for the opa-envoy-plugin PR to merge?

emaincourt commented:
Thanks ! Yes, hopefully my last changes will match the requirement. Once merged I’ll change the draft state of this PR.

emaincourt commented:
@srenatus The [pull request](https://github.com/open-policy-agent/opa-envoy-plugin/pull/323) was merged. I rebased this branch and squashed all the commits. I let you see if it still looks good to you. Thanks.
